CheckServicePrincipal/action.yml

```diff
@@ -26,75 +26,90 @@ runs:
          shell: pwsh
          id: get_expiry
          run: |
-              function RefreshToken($loginURL, $ClientID, $clientSecret, $tenantName) {
-                  $body = @{grant_type = "client_credentials"; client_id = $ClientID; client_secret = $ClientSecret; scope = $Env:SCOPE }
+              function RefreshToken($loginURL, $clientId, $clientSecret, $tenantName) {
+                  $body = @{grant_type = "client_credentials"; client_id = $clientId; client_secret = $clientSecret; scope = $env:SCOPE }
                   $oauthResponse = Invoke-RestMethod -Method POST -Uri $loginURL/$tenantName/oauth2/v2.0/token -Body $body
                   return $oauthResponse
               }
 
-              $Credentials = ( $Env:AZURE_CREDENTIALS | ConvertFrom-Json)
-              $body = @{grant_type = "client_credentials"; client_id = $Credentials.clientId; client_secret = $Credentials.clientSecret; scope = $Env:SCOPE }
-              $oauth = RefreshToken -loginURL $Env:LOGIN_URL  -resource $resource -ClientID $Credentials.clientId  -clientSecret $Credentials.clientSecret -tenantName $Env:TENNANT_NAME
+              $credentials = ( $env:AZURE_CREDENTIALS | ConvertFrom-Json )
+              $body = @{ grant_type = "client_credentials"; client_id = $credentials.clientId; client_secret = $credentials.clientSecret; scope = $env:SCOPE }
+              $oauth = RefreshToken -loginURL $env:LOGIN_URL  -resource $resource -clientId $credentials.clientId  -clientSecret $credentials.clientSecret -tenantName $env:TENNANT_NAME
               Write-Output "Connected with OAuth"
-              $headerParams = @{'Authorization' = "$($oauth.token_type) $($oauth.access_token)" }
-              $ApplicationsList = (Invoke-WebRequest -Headers $headerParams -Uri $Env:APP_SECRETS -Method GET)
-              $Logs = @()
-              $NextCounter = 0
-              do {
-                  foreach ($event in ($ApplicationsList.Content | ConvertFrom-Json | select -ExpandProperty value)) {
+              $headerParams = @{ 'Authorization' = "$($oauth.token_type) $($oauth.access_token)" }
+              $applicationsList = (Invoke-WebRequest -Headers $headerParams -Uri $env:APP_SECRETS -Method GET)
```

Comment:
Instead of returning the whole list of apps and iterating though them, could we return only the desired one?

See:
https://docs.microsoft.com/en-us/graph/api/application-list?view=graph-rest-1.0&tabs=http#optional-query-parameters
https://docs.microsoft.com/en-us/graph/query-parameters#search-parameter

Reply:
Addressed